kotlinpoet/src/test/java/com/squareup/kotlinpoet/MemberNameTest.kt

```diff
@@ -201,6 +201,42 @@ class MemberNameTest {
     )
   }
 
+  @Test fun importedMemberAndClassFunctionNameClash() {
+    val kotlinErrorMember = MemberName("kotlin", "error")
```

Comment:
I believe the issue you're facing is specific to `MemberName` and seems to not affect `TypeName`, so it might be worth digging into the differences of how imports work for both and fixing the behavior for `MemberName`. I tweaked your test a bit to use `ClassName` instead of `MemberName` and got it to pass with what looks like correct output:

```kotlin
@Test fun importedMemberAndClassFunctionNameClash() {
  val kotlinErrorType = ClassName("kotlin", "Error")
  val file = FileSpec.builder("com.squareup.tacos", "TacoTest")
    .addType(
      TypeSpec.classBuilder("TacoTest")
        .addFunction(
          FunSpec.builder("test")
            .addStatement("%T(%S)", kotlinErrorType, "errorText")
            .build()
        )
        .addType(
          TypeSpec
            .classBuilder("Error")
            .build()
        )
        .build()
    )
    .build()
  assertThat(file.toString()).isEqualTo(
    """
    |package com.squareup.tacos
    |
    |import kotlin.Unit
    |
    |public class TacoTest {
    |  public fun test(): Unit {
    |    kotlin.Error("errorText")
    |  }
    |
    |  public class Error
    |}
    |""".trimMargin()
  )
}
```